### PR TITLE
feat: website mode api

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### Issue

New / updated backend to support stepheader and stepfooter for website mode

### CODE REVIEW

File: apps/api-journeys/src/app/modules/journey/journey.graphql [hunk 1]
```diff
@@ -48,6 +59,13 @@ type Journey @key(fields: "id") {
   public title for viewers
   """
   displayTitle: String
+  showHosts: Boolean
+  showChatButtons: Boolean
+  showReactions: Boolean
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/journey/journey.graphql:64]:
```suggestion
  showReactionButtons: Boolean
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by levi-zustiak — CHANGES_REQUESTED** [review]

File: apps/api-gateway/schema.graphql [hunk 2]
```diff
@@ -1008,6 +1018,19 @@ input JourneyCreateInput
...
+enum JourneyMenuButtonIcon
+  @join__type(graph: JOURNEYS)
+{
+  menu1 @join__enumValue(graph: JOURNEYS)
+  equals @join__enumValue(graph: JOURNEYS)
+  home3 @join__enumValue(graph: JOURNEYS)
+  home4 @join__enumValue(graph: JOURNEYS)
+  more @join__enumValue(graph: JOURNEYS)
+  ellipsis @join__enumValue(graph: JOURNEYS)
+  globe01 @join__enumValue(graph: JOURNEYS)
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/api-gateway/schema.graphql:1030]: ahh yes you're right, i'll change this

File: apps/api-gateway/schema.graphql [hunk 3]
```diff
@@ -807,6 +807,7 @@ type ImageBlock implements Block
   Find a frontend implementation at https://github.com/woltapp/blurhash
   """
   blurhash: String!
+  scale: Int
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/api-gateway/schema.graphql:810]: Do we need the accuracy of a float? We've generally defaulted to using ints.

[the remark this answers is not in the document]
Comment (levi-zustiak) [reply apps/api-gateway/schema.graphql:810]: In my mind it's easier to store the scale value as a float if we're using the slider without any step values. Seems like there could be more calculations we have to perform to scale the value using an int. but it sounds like you already may have a general process

[the remark this answers is not in the document]
Comment (csiyang) [reply apps/api-gateway/schema.graphql:810]: We'll keep it as int, the mui slider uses ints from 0-100 by default too.

File: apps/api-journeys/db/schema.prisma [hunk 4]
```diff
@@ -508,6 +530,10 @@ model Block {
   parentBlock             Block?               @relation("ParentBlock", fields: [parentBlockId], references: [id], onDelete: Cascade)
   childBlocks             Block[]              @relation("ParentBlock")
   targetActions           Action[]             @relation("Block")
+  scale                   Int?
+  menuStepBlockParent     Journey?             @relation("MenuStepBlock")
+  logoImageBlockParent    Journey?             @relation("LogoImageBlock")
+  slug                    String?              @unique
```
Comment (csiyang) [inline apps/api-journeys/db/schema.prisma:536]: @tataihono I think this makes the field unique across the whole table right? Any ideas on how we could apply this per journey?

Comment (tataihono) [reply apps/api-journeys/db/schema.prisma:536]: drop this @unique req and put this further down
```
  @@unique([slug, journeyId])
```

**Review by tataihono — APPROVED** [review]

**Pull request merged**